### ISSUE TITLE
Page not found error when dismissing reports

### ISSUE THREAD

**Issue by MacilPrime** [savetheinternet/Tinyboard#71]
(mod-rewrite branch)

When the dismiss button is clicked on a report, you're given a page not found error, but if you go back, you can see that the report was successfully dismissed.

**Comment by savetheinternet** [comment]
I can't seem to reproduce this bug. All I get is [a 303 redirect and that's it](https://github.com/savetheinternet/Tinyboard/blob/master/inc/mod/pages.php#L1548), as intended.

**Comment by savetheinternet** [comment]
I'm going to close this issue for now. Feel free to reopen if the problem still exists.

**Comment by MacilPrime** [comment]
Figured it out. It only happens if additional_javascript_compile is on (this inconsistency is Issue 70 related) and post-hover.js is in use. When you hover over a link that's in a post body (that doesn't have rel="nofollow" set, like user posted links), post-hover.js tries to discern a post number from the link text, and if that post number isn't on the page, it makes an AJAX call to where the link points to. However, it has a bug in that if it fails to figure out the post number from the link text, it doesn't abort. There's no need for it to continue in that condition, because even if the link is to another thread, it doesn't know which post number it wants to show.

The dismiss report link was inside the post's tag on the report page, so when I hovered over the dismiss button a moment before clicking it, it had already been activated.

Starting at line 21 of post-hover.js:

``` javascript
        if(id = $link.text().match(/^>>(\d+)$/)) {
            id = id[1];
        }
```

An else block with a return statement like this fixes the bug:

``` javascript
        if(id = $link.text().match(/^>>(\d+)$/)) {
            id = id[1];
        } else {
            return;
        }
```
